packages/components/dashboard/inventory-card/inventory-card.tsx

```diff
@@ -0,0 +1,289 @@
+import * as React from 'react';
+import { useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+import { HealthBody } from '@openshift-console/dynamic-plugin-sdk-internal';
+import * as _ from 'lodash';
+import { useTranslation } from 'react-i18next';
+import { Link } from 'react-router-dom';
+import {
+  Gallery,
+  GalleryItem,
+  Card,
+  CardBody,
+  CardHeader,
+  CardTitle,
+  Divider,
+  Skeleton,
+} from '@patternfly/react-core';
+import {
+  BucketClassPhaseMap,
+  NamespaceStorePhaseMap,
+  ObjectBucketClaimPhaseMap,
+  PhaseType,
+} from '../../../constants';
+import {
+  BucketClassKind,
+  K8sResourceKind,
+  NamespaceStoreKind,
+} from '../../../types';
+import {
+  RedExclamationCircleIcon,
+  BlueInProgressIcon,
+} from '../../../utils/status/icons';
+import {
+  BucketClassPopOver,
+  DataResourcesPopOver,
+  OBCPopOver,
+} from '../../resource-pages/ResourcePopOver';
+import {
+  bucketClassResource,
+  nameSpaceStoreResource,
+  bucketClaimResource,
+} from '../../resources';
+import './inventory-card.scss';
+
+export const InventoryCard: React.FC = () => {
+  const { t } = useTranslation();
+  const [buckets, bucketsLoaded, bucketError] =
+    useK8sWatchResource<BucketClassKind[]>(bucketClassResource);
+  const [dataResources, dataResourcesLoaded, dataResourcesError] =
+    useK8sWatchResource<NamespaceStoreKind[]>(nameSpaceStoreResource);
+  const [obClaims, obClaimsLoaded, obClaimsError] =
+    useK8sWatchResource<K8sResourceKind[]>(bucketClaimResource);
+  type obcType = { name: string; ns: string };
+  const dataResourceErrorList = React.useMemo((): string[] => [], []);
+  const dataResourceProcessingList = React.useMemo((): string[] => [], []);
+  const bucketClassErrorList = React.useMemo((): string[] => [], []);
+  const bucketClassProcessingList = React.useMemo((): string[] => [], []);
+  const obcErrorList = React.useMemo((): [obcType] => [{} as obcType], []);
+  const obcProcessingList = React.useMemo((): [obcType] => [{} as obcType], []);
+
+  const bucketStatusMap: { [key: string]: number } = React.useMemo(
```

Comment:
```suggestion
  const [bucketsStatusMap, bucketPolicyErrorList, bucketPolicyProcessingList]: { [key: string]: number } = React.useMemo(
```

Reply:
calculate all three at once, in a single loop and return from this single useMemo...